packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -231,7 +230,18 @@ export class Dialog
     this.mutationObserver?.observe(this.el, { childList: true, subtree: true });
     connectLocalized(this);
     connectMessages(this);
-    connectFocusTrap(this);
+    connectFocusTrap(this, {
+      focusTrapOptions: {
+        clickOutsideDeactivates: false,
```

Comment:
Can you add a comment on why we set this to `false` here? Applies to modal. It might be helpful for other options too if it's not clear.